utxo/src/undo.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(unused, dead_code)]
+
+use crate::Utxo;
+use parity_scale_codec::{Decode, Encode};
+use common::primitives::BlockHeight;
+
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct TxUndo(Vec<Utxo>);
+
+impl TxUndo {
+    pub fn new_empty() -> Self {
+        Self(vec![])
+    }
+
+    pub fn new(utxos: Vec<Utxo>) -> Self {
+        Self(utxos)
+    }
+
+    pub fn push(&mut self, utxo: Utxo) {
+        self.0.push(utxo)
+    }
+
+    pub fn append(&mut self, other: &mut TxUndo) {
+        self.0.append(&mut other.0)
+    }
+
+    pub fn inner(&self) -> &[Utxo] {
+        &self.0
+    }
+
+    pub fn into_inner(self) -> Vec<Utxo> {
+        self.0
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct BlockUndo{
+    // determines at what height this undo file belongs to.
+    height:BlockHeight,
```

Comment:
Please make sure to format all the files before merging this.

Reply:
done.